docs/cookbook/manage-access-control-frontend-api.md

```diff
@@ -0,0 +1,78 @@
+# Managing Access Control in Frontend API
+
+This guide will walk you through the process of managing access control in the Frontend API. We will cover the default roles, how to grant permissions for queries/mutations, how to override these permissions, and how to use Symfony's voter system.
+
+## 1. Default Roles
+
+By default, we have three roles defined in `Shopsys\FrameworkBundle\Model\Customer\User\Role\CustomerUserRole`:
+
+-   `ROLE_API_LOGGED_CUSTOMER`: This role is assigned to a logged customer.
+-   `ROLE_API_ALL`: This role is assigned to a user with all privileges.
+-   `ROLE_API_CUSTOMER_SELF_MANAGE`: This role allows a user to manage only their own data.
+
+## 2. Granting Permissions for Query/Mutation
+
+Permissions for a query or mutation can be granted by defining access and using the `@=isGranted` directive in the query/mutation types files. For example:
+
+```yaml
+AddNewCustomerUser:
+    type: 'CustomerUser!'
+    description: 'Add new customer user to customer'
+    args:
+        input:
+            type: AddNewCustomerUserDataInput!
+            validation: cascade
+    access: "@=isGranted('ROLE_API_ALL')"
+    resolve: "@=mutation('addNewCustomerUserMutation', args, validator)"
+```
+
+## 3. Overriding Permissions
+
+Permissions can be overridden in the `project-base` to disable access for everyone or allow access for different roles. For example:
+
+```yaml
+AddNewCustomerUser:
+    type: 'CustomerUser!'
+    description: 'Add new customer user to customer'
+    args:
+        input:
+            type: AddNewCustomerUserDataInput!
+            validation: cascade
+    access: false
+    resolve: "@=mutation('addNewCustomerUserMutation', args, validator)"
+```
+
+## 4. Symfony Voter
+
+Symfony's voter system is a powerful tool for managing access control. There is already a defined voter `can_manage_customer_user` that can be used to check if a user has the necessary permissions to manage a customer user. The voter checks the user's roles and returns either `ACCESS_GRANTED`, `ACCESS_DENIED`, or `ACCESS_ABSTAIN`.
+
+### 4.1 Voter Implementation
+
+The `can_manage_customer_user` voter is implemented in `Shopsys\FrontendApiBundle\Voter\CustomerUserVoter`. This voter extends `Shopsys\FrontendApiBundle\Voter\AbstractB2bVoter`, which by default only allows access for B2B domains. For more information on setting up domain type, refer to the [Shopsys documentation](https://docs.shopsys.com/en/14.0/introduction/start-building-your-application/#set-up-domain-type).
+
+### 4.2 Overriding the Voter
+
+If you want to modify voter logic you can simply override `Shopsys\FrontendApiBundle\Voter\CustomerUserVoter` or `Shopsys\FrontendApiBundle\Voter\AbstractB2bVoter` to modify the access control logic or disable it entirely.
```

Comment:
Can I override the abstract class and get a new behavior with it?

Reply:
Good point, no you can't. Updated